packages/ubuntu_desktop_installer/lib/pages/keyboard_layout/keyboard_layout_detector.dart

```diff
@@ -0,0 +1,227 @@
+import 'dart:convert';
+
+/// Detects keyboard layout according to the steps defined in
+/// `/usr/share/console-setup/pc105.tree`.
```

Comment:
Nit: This class parses an arbitrary file, not necessarily the one specified here. The .tree file is installed as part of the installer, and the system one is not used anyway.